### PR TITLE
[#189] Event Processor section update

### LINKED ISSUE #189: Pooled Tracking Event Processor section
Framework issue [#1712](https://github.com/AxonFramework/AxonFramework/pull/1712) has seen the introduction of a new type of `EventProcessor`. Namely, the `PooledTrackingEventProcessor`. 
The [Evnet Processors](https://docs.axoniq.io/reference-guide/axon-framework/events/event-processors) page should receive a description on what this implementation does differently (and similar) with the other processors.

As a first stab, a short paragraph would be sufficient. 
The reasoning behind this is because, in essence, the entire "Event Processors" page deserves a rewrite.
Either this issue contains the entire rewrite or a simple paragraph addition to the existing page.

### PR DESCRIPTION
**Pull request by smcvb**
This pull request updates the entire event processor section of the reference guide.
First and foremost, the single `event-processor.md` page is replaced by three files:

- `README.md` - the mean section introducing the Event Processor and the common configuration among all Event Processor implementations
- `subscribing.md` - a dedicated page towards the Subscribing Event Processor, its specifics, and how it is configured
- `streaming.md` - a dedicated page towards the Streaming Event Processor, its specifics, and how it is configured

In this process, quite some specifics have been added that weren't described in this detail yet.
Things like:
- Assignment rules on Event Handling Components to Event Processors
- Generic Event Processor configuration options, like error handlers, interceptors, transaction managers, etc
- An introduction to what the Subscribing Event Processor is
- An introduction to what the Streaming Event Processor is
- The difference between the Tracking Event Processor and the Pooled Streaming Event Processor
- Dedicated configuration sections for TEPs and PSEPs
- More details on what's configurable around tokens, like claim timeouts, token store specifics, and initial tokens
- Expansion on the specifics of parallel processing, the threading differences between the TEP and PSEP on this front, and what can be configured to go for parallel processing
- More information and samples on how to split, merge or release (for load balancing in a multi-node scenario) segments
- Expanded the Replay Events section by exhausting the reset token operations and going into more detail

Also, some sections have been moved from the old `event-processor.md` file to other spots, like:
- The notice on distributing events is added to the `event-bus-and-event-store.md`
- The Saga configuration section is moved to the Saga pages

Lastly, any headers that were used as links have been renamed.
This increased the number of changed files for this pull request.

This pull request resolves #189 but expands on it quite some by completely adjusting the Event Processor pages.

### CODE REVIEW

File: axon-framework/sagas/implementation.md [hunk 1]
```diff
@@ -129,3 +102,103 @@ To configure caching, simply wrap any `SagaStore` in a `CachingSagaStore`. The c
...
+class MySaga {
+    // saga implementation left out...
+}
+```
+{% endtab %}
+{% endtabs %}
+
+Although the defaults lead us to a working Saga environment, it is recommended to define the [`SagaStore`](#saga-repository-and-saga-store) to use.
+The `SagaStore` represents the mechanism that 'physically' stores the Saga instances, for which it uses the `AnnotatedSagaRepository` \(the default\) to store and retrieve Saga instances.
+Providing no storage means for a Saga means Axon defaults to an `InMemorySagaStore`, thus not persisting the Saga on shutdown.
```
Comment (brunchey) [inline axon-framework/sagas/implementation.md:154]: Maybe change 
Providing no storage means for a Saga means Axon defaults to...
to...
If no `SagaStore` is configured, Axon defaults to....

**Review by brunchey — APPROVED** [review]

File: axon-framework/events/event-processors/subscribing.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
+# Subscribing Event Processor
+
+The `SubscribingEventProcessor`, or Subscribing Processor for short, is a type of [Event Processor](README.md).
+As any Event Processor, it serves as the technical aspect to handle events by invoking the event handlers written in an Axon application.
+
+The Subscribing Processor defines itself by receiving the events from a `SubscibableMessageSource`.
```
[the remark this answers is not in the document]
Comment (smcvb) [reply axon-framework/events/event-processors/subscribing.md:6]: Good spot, thanks!

File: axon-framework/events/event-bus-and-event-store.md [hunk 3]
```diff
@@ -380,3 +380,16 @@ It is possible to use a different serializer for the storage of events, than all
 
 If no explicit `eventSerializer` is configured, events are serialized using the main serializer that has been configured \(which defaults to the `XStreamSerializer`\).
 
+## Distributing Events
+
+The use of an `EventStore` provides a simple way towards distributing events: by sharing the data source.
```
Comment (SaraTorrey) [inline axon-framework/events/event-bus-and-event-store.md:385]:
```suggestion
The use of an `EventStore` provides a simple way of distributing events: by sharing the data source.
```

File: axon-framework/events/event-bus-and-event-store.md [hunk 4]
```diff
@@ -380,3 +380,16 @@ It is possible to use a different serializer for the storage of events, than all
...
+This is the approach taken with [Axon Server](../../axon-server-introduction.md), for example.
```
Comment (SaraTorrey) [inline axon-framework/events/event-bus-and-event-store.md:386]:
```suggestion
This is the approach taken by [Axon Server](../../axon-server-introduction.md), for example.
```

File: axon-framework/events/event-processors/subscribing.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+The Subscribing Processor defines itself by receiving the events from a `SubscribableMessageSource`.
+The `SubscribableMessageSource` is an infrastructure component to register a Subscribing Processor too.
+
+After registration to the `SubscribableMessageSource`, the message source gives the events to the `SubscribingEventProcessor` in the order they are received.
+Examples of a `SubscribableMessageSource` are the `EventBus` or the [AMQP Extension](../../../extensions/spring-amqp.md).
+Both the `EventBus` and AMQP Extension are simple message bus solutions for events.
+
+The simple bus solution makes the `SubscribableMessageSource` and thus the Subscribing Processor an approach to only receive _current_ events.
+Operations like [replaying](streaming.md#replaying-events) are thus not an option for any Subscribing Processor as long as the `SubscribableMessageSource` follows this paradigm.
```
Comment (SaraTorrey) [inline axon-framework/events/event-processors/subscribing.md:14]:
```suggestion
Operations like [replaying](streaming.md#replaying-events) are, therefore, not an option for any Subscribing Processor as long as the `SubscribableMessageSource` follows this paradigm.
```

File: axon-framework/events/event-processors/subscribing.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+Furthermore, the message source will use the same thread that receives the events to invoke the registered Subscribing Processors.
+When the `EventBus` is, for example, used as the message source, this means that the event publishing thread is the same one handling the event in the Subscribing Processor.
+
+Although this approach deserves a spot within the framework, most scenarios require further decoupling of components by separating the threads as well.
+When, for example, an application requires event processing parallelization to get a higher performance, this can be a blocker.
+This predicament is why the `SubscribingEventProcessor` is not the default within Axon Framework.
```
Comment (SaraTorrey) [inline axon-framework/events/event-processors/subscribing.md:21]:
```suggestion
This predicament is why the `SubscribingEventProcessor` is not the default in Axon Framework.
```

**Review by SaraTorrey — APPROVED** [review]

**Pull request merged**